controller/src/beerocks/master/db/node.cpp

```diff
@@ -297,3 +297,47 @@ bool node::set_type(beerocks::eType type_)
     }
     return false;
 }
+
+bool node::link_metrics_data::add_transmitter_link_metric(
+    std::shared_ptr<ieee1905_1::tlvTransmitterLinkMetric> TxLinkMetricData)
+{
+    size_t info_size = TxLinkMetricData->interface_pair_info_length();
+
+    for (size_t i = 0;
+         i < info_size / sizeof(ieee1905_1::tlvTransmitterLinkMetric::sInterfacePairInfo); i++) {
```

Comment:
You are right, I can see now that on `interface_pair_info_length()` implementation, the function returns `num_of_element * sizeof(sInterfacePairInfo)`. It is not so clear to me why it does it, instead of returning num of elements, but oh well.
 Anyway, if it confused me, it will probably confuse more people, so I suggest to do make it clear with comments:
```cpp
//  interface_pair_info_length() returns the length in bytes (number of elements * sizeof(sInterfacePairInfo).
size_t info_length_bytes = TxLinkMetricData->interface_pair_info_length();
auto info_length_elements = info_length_bytes / sizeof(ieee1905_1::tlvTransmitterLinkMetric::sInterfacePairInfo);

for (size_t i = 0; i < info_length_elements ; i++) {
   // code
}
```